docs/user/rolling-out.rst

```diff
@@ -174,3 +174,155 @@ And confirm that Shipper believes this rollout to be done:
+
+*******************
+RolloutBlock object
+*******************
+
+Here's an example for a RolloutBlock object we'll use:
+
+.. code-block:: yaml
+    apiVersion: shipper.booking.com/v1alpha1
+    kind: RolloutBlock
+    metadata:
+      name: dns-outage
+      namespace: rollout-blocks-global # for global rollout block. for a local one use the correct namespace.
+    spec:
+      message: DNS issues, troubleshooting in progress
+      author:
+        type: user
+        name: jdoe # This indicates that a rollout block was put in place by user 'jdoe'
+
+Copy this to a file called ``globalRolloutBlock.yaml`` and apply it to our Kubernetes cluster:
+
+.. code-block:: shell
+
+    $ kubectl apply -f globalRolloutBlock.yaml
+
+This will create a Global *RolloutBlock* object.
+In order to create a namespace rollout block, simply state the relevant namespace in the yaml file.
+
+While this object is in the system, there can not be any change to the `.Spec` of any object. Shipper
+will reject the creation of new objects and patching of existing releases.
+
+**************************
+Overriding a rollout block
+**************************
+
+Rollout blocks can be overriden with an annotation applied to the *Application* or *Release* object which
+needs to bypass the block. This annotation will list each RolloutBlock object that it overrides with
+a fully-qualified name (namespace + name).
+
+For example, mending our Application object to override
+the global rollout block that we set in place:
+
+.. code-block:: yaml
+
+  apiVersion: shipper.booking.com/v1alpha1
+  kind: Application
+  metadata:
+    name: super-server
+    annotations:
+      shipper.booking.com/block.override: rollout-blocks-global/dns-outage
+  spec:
+    revisionHistoryLimit: 3
+    template:
+      # ... rest of template omitted here
+
+The annotation may reference multiple blocks:
+
+.. code-block:: yaml
+
+    shipper.booking.com/block.override: rollout-blocks-global/dns-outage,frontend/demo-to-investors-in-progress
+
+The block override annotation format is CSV.
+
+The override annotation **must** reference specific, fully-qualified *RolloutBlock* objects by name.
+Non-existing blocks enlisted in this annotation are not allowed.
+If there exists a Release object for a specific application, the release should be the one overriding.
+
+**********************************
+Application and Release conditions
+**********************************
+
+Application and Release objects will have a `.status.conditions` entry which lists all of the
+blocks which are currently in effect.
+
+For example:
+
+.. code-block:: yaml
+    apiVersion: shipper.booking.com/v1
+    kind: Application
+    metadata:
+      name: ui
+      namespace: frontend
+    spec:
+      # ... spec omitted
+    status:
+      conditions:
+      - type: RolloutBlock
+        status: True
+        reason: rollout-blocks-global/dns-outage
```

Comment:
[K8s conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status) say this about `reason`:

>In condition types, and everywhere else they appear in the API, Reason is intended to be a one-word, CamelCase representation of the category of cause of the current status


On the other hand, saying that a `RolloutBlock` reason is... `RolloutBlock`... it's a bit weird, so I wonder if this is the correct name for this condition.